.github/workflows/actions.yml

```diff
@@ -3,11 +3,7 @@
 
 name: JUnit Pioneer Flow
 
-on:
-  push:
-    branches: [ master ]
-  pull_request:
-    branches: '*'
+on: [push, pull_request]
```

Comment:
i do not think that we really need a build on each push to all branches. `master` alone is fine. Based on shipkit we also do not want build on tags i think.

Reply:
What is the downside of building on each push? If we don't touch the parallelization limit such that jobs are queued/blocked, I would personally find this helpful as I cannot locally verify the build matrix. As a contributor, I wish to get that feedback _before_ opening a PR.

Regarding Shipkit: unfortunately, I haven't really worked with it so far. This might be an issue because don't know how these CI skips work (neither in general nor in combination with GHA).

I would suggest to revert this commit and discuss that separately, maybe also in the course of #193.

Reply:
I have to admit, i did not take the contributor part into consideration, but, i am also thinking about resource usage etc. and i am not sure, if we really need a build on 3 jdks with 3 OS for a branch in development.

The thing we really should take care of, is to not build on a _tag_ push. Shipkit will create a tag and push it, hence that the build on master will trigger a another build, which is imho not needed at all. Hence that i would exclude tags.

`[ci skip]` is not working on GHA as it is a travis default
`[ci skip-release]` is something [offered by skipkit ](https://github.com/mockito/shipkit/blob/master/docs/features/merge-pull-requests.md#user-guide)

we could although configure our own "commit postfix" to prevent a release - to make it easier for @nicolaiparlog to not trigger a build :)